R/read-eval.R

```diff
@@ -0,0 +1,176 @@
+#' Parse R code.
+#'
+#' These functions parse_expr and transform text into R expressions.
+#' This is the first step of interpreting or evaluating a piece of
+#' code written by a programmer.
+#'
+#' \code{parse_expr()} returns one expression. If the text contains
+#' more than one expression (separated by colons or new lines), an
+#' error is issued. On the other hand \code{parse_exprs()} can handle
+#' multiple expressions. It always returns a list of expressions
+#' (compare to \code{\link[base]{parse}()} which returns an obsolete
+#' expression vector). All functions also support R connections (and
+#' will close the connection after parsing).
+#'
+#' The versions prefixed with \code{f_} return expressions quoted in
+#' formulas rather than raw expressions.
+#'
+#' @param x Text containing expressions to parse_expr for
+#'   \code{parse_expr()} and \code{parse_exprs()}. Can also be an R
+#'   connection, for instance to a file. Note that the connection will
+#'   be closed as side effect.
+#' @param env The environment for the formulas. Defaults to the
+#'   context in which the parse_expr function was called. Can be any object
+#'   with a \code{as_env()} method.
+#' @return \code{parse_expr()} returns a formula, \code{parse_exprs()}
+#'   returns a list of formulas.
+#' @seealso \code{\link[base]{parse}()}
+#' @export
+#' @examples
+#' # parse_expr() can parse_expr any R expression:
+#' parse_expr("mtcars %>% dplyr::mutate(cyl_prime = cyl / sd(cyl))")
+#'
+#' # A string can contain several expressions separated by ; or \n
+#' parse_exprs("NULL; list()\n foo(bar)")
+#'
+#' # The versions prefixed with f_ return formulas:
+#' f_parse_expr("foo %>% bar()")
```

Comment:
You're right. It now occurs to me it would be more consistent if `f_new()` was renamed to `new_f()`, since it seems the constructors in the tidyverse use `new_` as prefix. What do you think?